switchmap/server/db/db.py

```
@@ -286,3 +286,48 @@ def db_add_all(error_code, instances, die=True):
+        error_code: Error code to use in messages
+        instances: List of instances
+        die: Whether to raise an exception on failure
+
+    Returns:
+        result: True if the transaction is successful
+    """
+    # Initialize key variables
+    result = False
+
+    with ENGINE.connect() as connection:
+        with Session(bind=connection, future=True) as session:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)